bundle/bundle.go

```diff
@@ -594,6 +599,10 @@ func (r *Reader) Read() (Bundle, error) {
 			if err := util.NewJSONDecoder(&buf).Decode(&bundle.Manifest); err != nil {
 				return bundle, fmt.Errorf("bundle load failed on manifest decode: %w", err)
 			}
+			// Prefix Wasm module file path with root dir path, should we be reading a directory and not a tarball
+			for i, wr := range bundle.Manifest.WasmResolvers {
+				bundle.Manifest.WasmResolvers[i].Module = r.loader.FilePrefix() + wr.Module
```

Comment:
Can you please elaborate why we would need this? The doc says:
```
module - A string path to the Wasm module relative to the root of the bundle.
```
So is this not already taken into account? 

If we do, can we use `r.baseDir` or handle this during bundle creation?

Reply:
The problem solved here is when reading a bundle from a folder, and not a tarball. In such case, we'll end up not finding any Wasm module files as the files all have absolute paths, and the `.manifest` list them with paths relative to the tarball root (see `bundle/bundle.go:279`). 

We can't solve this at build-time, as it's likely the location of the Wasm files aren't the same when building a bundle as when consuming it.

Reply:
So you're saying this is currently broken for the directory use-case? 

When reading bundles from directories, `opa build` will set `r.baseDir` with the directory path. Can we use that? https://sourcegraph.com/github.com/open-policy-agent/opa@ea06d89ea73650d13d33a832b156ca7d308a70c4/-/blob/bundle/bundle.go?L711

Reply:
`r.baseDir` seems to contain the information needed 👍 . 
While making the change, I encountered a test that started failing: https://github.com/open-policy-agent/opa/blob/35b2b6dace8c696eee023e403b1f35210f317c0c/bundle/bundle_test.go#L165. This test looks to be asserting that the _base dir_ is injected into the manifest at build time, which clashes with the updated code. Given that this test is asserting a behavior I don't understand, I'm bowing out from this drive-by fix, and will update the tests to not read bundles from a dir. I'll create a bug ticket describing the issue, so that it can be tracked and solved separately, and won't hold up this PR.

Reply:
https://github.com/open-policy-agent/opa/issues/5620